design/sequence/scheduler-low.puml

```diff
@@ -0,0 +1,108 @@
+@startuml
+
+box "server"
+    participant ":Database" as db
+    participant ":Scheduler" as s
+    participant "j:Job" as j
+    participant ":Dispatcher" as d
+end box
+
+s++
+s -> db++: get_jobs()
+db --> s--: jobs: list[Job]
+
+s -> s++: sorted(jobs)
+note right of s
+    Sorts jobs list by effective priority.
+end note
+s --> s--: jobs
+
+s -> db++: get_available_workers()
+note right of db
+    Return workers with free resources.
+end note
+db --> s--: available_wcs: list[WorkMachine]
+
+loop j in jobs
+    
+    s->s++ : preemption_threshold()
+    s-->s-- : preemption : float
+    
+    alt j.get_eff_priority > preemption
+        s -> db++ : lowest_priority_wm()
+        note right of s
+            Returns WorkMachine
+            whose jobs have the
+            lowest summed priority.
+        end note
+        db --> s--: wm: WorkMachine
+        s -> s ++: pause(wm, j)
+        note right of s
+            Pauses enough jobs
+            on wm to run j.
+        end note
+        
+        s--> s--
+
+        s -> j++: set_wm(wm)
+        j --> s--
+    
+    s->s++ : blocking_threshold()
+    s-->s-- : blocking : float
```

Comment:
The program logic in the diagram is broken for the above two lines.
In the diagram they only get executed in the first alt block but they are needed outside of it.